File: LeetCode/Medium/153. Find Minimum in Rotated Sorted Array.php
class Solution
{



    function findMin( $nums, $i = 0, $j = null )
    {
        if ( $j === null )
        {
            $j = count( $nums ) - 1;
        }
        
        if ( $nums[$i] <= $nums[$j] )
        {
            return $nums[$i];
        }

        $k = floor( ( $i + $j ) / 2 );
        
        if ( $nums[$i] <= $nums[$k] )
        {
            return $this->findMin( $nums, $k + 1, $j );
        }
        else
        {
            return $this->findMin( $nums, $i, $k );
        }
    }



}

File: LeetCode/Medium/153. Find Minimum in Rotated Sorted Array.java
class Solution
{
    public int findMin( int[] nums )
    {
        int i = 0;
        int j = nums.length - 1;
        int k = 0;
        
        while ( i < j )
        {
            k = ( i + j ) / 2;
            
            if ( nums[0] > nums[k] )
            {
                j = k;
            }
            else
            {
                i = k + 1;
            }
        }
        
        if ( i == nums.length - 1 )
        {
            return Math.min( nums[0], nums[i] );
        }
        else
        {
            return nums[i];
        }
    }
}
